### PR TITLE
update ci documentation

### PR DESCRIPTION
**Pull request by Okhelifi**
… developments [skip ci]

## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

A small description of what you did in that PR.

### CODE REVIEW

File: .circleci/ci/README.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ APIM CI run in two steps:
 1. A pre-step will run the `generate-config` job generate from typescript code the circleCI workflow to execute. (via `npm run generate /tmp/dynamicConfig.yml`)
 2. The CircleCI orbs `continuation/continue` execute the generated workflow.
 
+### Conditions to generate the pipeline
+The pipeline generation is available according to different conditions:
+- if the branch is master ( CIRCLE_BRANCH is master )
+- if we are working on a specific tag ( CIRCLE_TAG is defined )
```
Comment (phiz71) [inline .circleci/ci/README.md]: Not for the moment.
In APIM, we don't use the TAG yet. Only the branch.

**Review by jourdiw — DISMISSED** [review]

**Review by jourdiw — APPROVED** [review]
LGTM

**Comment by phiz71** [comment]
I think we should consider support branches like master branch. I mean, a push on a support branch should build everything.

File: .circleci/ci/src/index.ts [hunk 2]
```diff
@@ -20,11 +19,16 @@ if (isBlank(CIRCLE_SHA1)) {
...
+ * The pipeline generation is available according to different conditions:
+ *     - if the branch is supported ( CIRCLE_BRANCH is master or a support branch )
+ *     - if we are working on a branch with changes committed on the base branch
+ */
+const changed = isSupportBranchOrMaster(CIRCLE_BRANCH) ? Promise.resolve([]) : changedFiles(`origin/${GIT_BASE_BRANCH}`);
 
 changed
   .then((changes) => ({
-    branch: CIRCLE_BRANCH ?? CIRCLE_TAG ?? 'unknown',
+    branch: CIRCLE_BRANCH ?? 'unknown',
```
Comment (jourdiw) [inline .circleci/ci/src/index.ts]: Unnecessary `?? 'unknown'` because you set a default value above on line 8.

**Pull request merged**